pinctrl/digital_interrupts.go

```diff
@@ -17,20 +17,25 @@ import (
 
 // DigitalInterrupt is the struct for managing a digital interrupt that satisfies a board.DigitalInterrupt.
 type DigitalInterrupt struct {
-	workers  *utils.StoppableWorkers
-	line     *gpio.LineWithEvent
-	mu       sync.Mutex // Protects everything below here
-	config   board.DigitalInterruptConfig
-	count    int64
-	channels []chan board.Tick
+	workers             *utils.StoppableWorkers
+	line                *gpio.LineWithEvent
+	mu                  sync.Mutex // Protects everything below here
+	config              board.DigitalInterruptConfig
+	count               int64
+	channels            []chan board.Tick
+	lastEvent           int64 // last time the event was triggered
+	debounceNanoSeconds int64
 }
 
 // NewDigitalInterrupt constructs a new digitalInterrupt from the config and pinMapping. If
 // oldInterrupt is not nil, all channels added to it are added to the new interrupt and removed
 // from the old one.
+// debounceMilliSeconds allows users to set a software debounce timer for their interrupt.
+// Setting this to 0 disables the feature.
 func (ctrl *Pinctrl) NewDigitalInterrupt(
 	config board.DigitalInterruptConfig,
 	pinMapping genericlinux.GPIOBoardMapping,
+	debounceMilliSeconds int,
```

Comment:
why have it as milliseconds here and nanoseconds everywhere else? is milliseconds the standard for debounce timers?

Reply:
milliseconds is what we used for the pin config in the [scope](https://docs.google.com/document/d/1tMCwgCXLuA6EIoI0DQS3VWbCFlQPrajgzTZmUl22Vwg/edit?tab=t.0#heading=h.exsceggwzbwr) and milliseconds is generally going to be easier for a human to think about. 

I agree its kinda weird, and in the pi4 implementation its actually worse because that implementation uses milliseconds, microseconds, and nanoseconds because of how the timing is defined for the interrupts